akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -137,18 +115,17 @@ trait LowPriorityToResponseMarshallerImplicits {
                   case Marshalling.WithFixedContentType(`best`, marshal) ⇒ marshal
                   case Marshalling.WithOpenCharset(`bestMT`, marshal)    ⇒ () ⇒ marshal(bestCS)
                 }
-            }).toList
+            }
 
             // TODO we could either special case for certrain known types,
             // or extend the entity support to be more advanced such that it would negotiate the element content type it
             // is able to render.
-            if (selectedMarshallings.isEmpty) throw new NoStrictlyCompatibleElementMarshallingAvailableException[T](s.contentType, marshallings)
```

Comment:
Was it intentional that this logic (introduced by https://github.com/akka/akka-http/commit/238e7fb) should apply to `fromEntityStreamingSupportAndEntityMarshaller` but not `fromEntityStreamingSupportAndByteStringMarshaller`?